### ISSUE TITLE
Quota module/attribute for chests to enforce stricter anti-grief measures

### ISSUE THREAD

**Issue by Palmidence** [Electroid/PGM#344]
I'm proposing an attribute/module for PGM which would make it harder for players to take a bunch of item stacks from chests on resource-dependent maps (such as Spaceship Battles and older maps) which severely hinders team progress. This would help self-moderate the players and self-regulate the match.

The implementation could be as simple as a limit on the amount they can take from a chest (which would apply to that player's life and slow down resource hoarding), or a cooldown for how many of that item stack a player can take out from the chest. This would allow for things such as one player only being allowed to take 2 stacks of TNT per life or 2 stacks of logs every 5 minutes.

This is the only method I can think of which prevents the ever-lasting issue of resource hoarding or making games stalemate quicker because of a lack of supplies. As we're discussing ways to end matches quicker in general, I think this would help with matches on older maps so that there are more supplies to work with and it doesn't come to a grinding halt or one team at an entire disadvantage due to a lack of resources.

**Comment by KingOfSquares** [comment]
But this would have to be defined per map right, so a XML tag?

To prevent jankyness with CTW and special resources on some maps (Example : Power Bows) this would likely also be a kind of region filter. 

So in practice it would define a limit of items* a player can extract from chests in a specified region each death/minute/hour etc.

* = Defaults to all items, optionally you can choose which item(s) the filter would apply to.

**Comment by Palmidence** [comment]
@KingOfSquares Yes, this would be an XML tag, and would likely be used as a region filter (because wool regenerates and something like various potions may be required to leave a well-defended room.) You hit the nail on the head with how it'd work.
